### PR TITLE
Add ServerRoom.MustHaveMembershipForUser

### LINKED ISSUE #137: federation.ServerRoom should have CheckMembershipForUser
See https://github.com/matrix-org/complement/pull/131/files

### PR DESCRIPTION
**Pull request by kegsay**
As per https://github.com/matrix-org/complement/pull/131/files#r658674594

Fixes #137

### CODE REVIEW

File: internal/federation/server_room.go [hunk 1]
```diff
@@ -112,6 +113,21 @@ func (r *ServerRoom) AuthChain() (chain []*gomatrixserverlib.Event) {
 	return
 }
 
+func (r *ServerRoom) MustHaveMembershipForUser(t *testing.T, userID, wantMembership string) {
```
Comment (richvdh) [inline internal/federation/server_room.go:116]: a brief doc comment might not go amiss, even if it's a bit obvious.

**Review by richvdh — APPROVED** [review]
it feels like a bit of a shame to lose the context about when the test is happening - "after invite" / "after reject", but I'll not fight for it.

Thanks!

**Comment by kegsay** [comment]
You don't lose context because this is using `t.Helper()` so the file/line number will be the stack frame above. In other words you'll see for example:
```
federation_room_invite_test.go:54: no membership state for @alice:localhost
federation_room_invite_test.go:60: no membership state for @alice:localhost
```
and the surrounding code at those line numbers give you the context of after invite / after reject.

**Comment by richvdh** [comment]
> You don't lose context because this is using `t.Helper()` so the file/line number will be the stack frame above. In other words you'll see for example:
> 
> ```
> federation_room_invite_test.go:54: no membership state for @alice:localhost
> federation_room_invite_test.go:60: no membership state for @alice:localhost
> ```

it'd be nice to know what the context is without having to correlate line numbers, but I agree, this helps.

**Pull request merged**